uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/controller/UniProtKBByAccessionControllerIT.java

```diff
@@ -397,6 +397,147 @@ void searchForDeletedInactiveEntriesReturnItself() throws Exception {
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.SEE_OTHER.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON_VALUE))
+                .andExpect(header().string(HttpHeaders.LOCATION, "/uniprotkb/P21802?from=FGFR2_HUMAN"));
+    }
+
+    @Test
+    void searchWithIDForMergedInactiveEntriesRedirectToAccession() throws Exception {
+        // given
+        UniProtKBEntry entry = UniProtEntryMocker.create(UniProtEntryMocker.Type.SP_CANONICAL);
+        getStoreManager().save(DataStoreManager.StoreType.UNIPROT, entry);
+
+        List<InactiveUniProtEntry> mergedList =
+                InactiveEntryMocker.create(InactiveEntryMocker.InactiveType.MERGED);
+        getStoreManager()
+                .saveEntriesInSolr(DataStoreManager.StoreType.INACTIVE_UNIPROT, mergedList);
+
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                get(ACCESSION_RESOURCE, "Q14301_FGFR2")
+                                        .header(ACCEPT, APPLICATION_JSON_VALUE));
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.SEE_OTHER.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON_VALUE))
+                .andExpect(header().string(HttpHeaders.LOCATION, "/uniprotkb/Q14301?from=Q14301_FGFR2"));
+    }
+
+    @Test
+    void searchWithIDForDeletedInactiveEntriesRedirectToAccession() throws Exception {
+        // given
+        UniProtKBEntry entry = UniProtEntryMocker.create(UniProtEntryMocker.Type.SP_CANONICAL);
+        getStoreManager().save(DataStoreManager.StoreType.UNIPROT, entry);
+
+        List<InactiveUniProtEntry> deletedList =
+                InactiveEntryMocker.create(InactiveEntryMocker.InactiveType.DELETED);
+        getStoreManager()
+                .saveEntriesInSolr(DataStoreManager.StoreType.INACTIVE_UNIPROT, deletedList);
+
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                get(ACCESSION_RESOURCE, "I8FBX2_YERPE")
+                                        .header(ACCEPT, APPLICATION_JSON_VALUE));
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.SEE_OTHER.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON_VALUE))
+                .andExpect(header().string(HttpHeaders.LOCATION, "/uniprotkb/I8FBX2?from=I8FBX2_YERPE"));
+    }
+
     @Override
```

Comment:
It would be good if we could add more negative test cases.
1. P12345..txt (regex should catch it)
2. redirect for other formats (e.g. xml)
3. when one protein id is mapped to more than one entries
4. some non-existing version

Reply:
1. Good point, I will create a UnitTest for the regex
2. This is other code, right? (probably covered by other test) In the controller we receive the correct MediaType and use it. Does it make sense?
3. It is covered in the Merged and Demerged integration test
4. I think this is other code as well, right? (unisave code, right?)

Reply:
I just pushed the unit tests for FieldRegexConstants (FieldRegexConstantsTest)